apps/transmitmedia.cpp

```diff
@@ -204,22 +234,33 @@ static void PrintSrtStats(int sid, const CBytePerfMon& mon, ostream &out)
         output << "WINDOW      FLOW: " << setw(11) << mon.pktFlowWindow      << "  CONGESTION: " << setw(11) << mon.pktCongestionWindow  << "  FLIGHT: " << setw(11) << mon.pktFlightSize << endl;
         output << "LINK         RTT: " << setw(9)  << mon.msRTT            << "ms  BANDWIDTH:  " << setw(7)  << mon.mbpsBandwidth    << "Mb/s " << endl;
         output << "BUFFERLEFT:  SND: " << setw(11) << mon.byteAvailSndBuf    << "  RCV:        " << setw(11) << mon.byteAvailRcvBuf      << endl;
-    }
-
-    out << output.str() << std::flush;
-}
+        return output.str();
+    } 
 
-static void PrintSrtBandwidth(double mbpsBandwidth)
-{
-    std::ostringstream output;
-
-    if (printformat == PRINT_FORMAT_JSON) {
-        output << "{\"bandwidth\":" << mbpsBandwidth << '}' << endl;
-    } else {
+    string WriteBandwidth(double mbpsBandwidth) override 
+    {
+        std::ostringstream output;
         output << "+++/+++SRT BANDWIDTH: " << mbpsBandwidth << endl;
+        return output.str();
     }
+};
 
-    cerr << output.str() << std::flush;
+shared_ptr<SrtStatsWriter> SrtStatsWriterFactory(PrintFormat printformat)
+{
+    switch (printformat)
+    {
+    case PRINT_FORMAT_JSON:
+        return make_shared<SrtStatsJson>();
+        break;
+    case PRINT_FORMAT_CSV:
+        return make_shared<SrtStatsCsv>();
+        break;
+    case PRINT_FORMAT_2COLS:
+        return make_shared<SrtStatsCols>();
+        break;
+    default:
+        return nullptr;
```

Comment:
Although I missed one point.
`stats_writer = nullptr` is not checked further in the code.

Reply:
Good catch! Fixed.